sendgrid/helpers/mail/mail.py

```diff
@@ -703,15 +703,11 @@ def get(self):
 
 class ASM(object):
 
-    def __init__(self, group_id=None, groups_to_display=None):
-        self._group_id = None
-        self._groups_to_display = None
-
-        if group_id is not None:
-            self._group_id = group_id
-
-        if groups_to_display is not None:
-            self._groups_to_display = groups_to_display
+    def __init__(self, group_id, groups_to_display=None):
```

Comment:
What do you think about changing L710 to be

```
self.groups_to_display = groups_to_display
```

and then moving the validation and exception to [line 724](https://github.com/sendgrid/sendgrid-python/pull/413/files#diff-8f6b078362cfbc79d2a0f6a3f1914910R724) ? 

I think that this would allow any setting of the groups to be handled in the same way, whether by __init__ or otherwise.

Reply:
Should we also add logic to group_id that causes a similar exception if it is turned on and there are more than 25 groups?

Reply:
If you're OK with doing error-checking in Mail I'll do that and look for other API limits that we can verify.

Reply:
I think that it makes sense to try to catch the limitations in the natural and earliest bottlenecks. That way, the code fails quickly, rather than after running all the way through and then failing.

Reply:
Oh I mean like in the Mail helper and the small classes (e.g. Personalization), not just the Mail class.  I saw that big comment about leaving validation up to the user so I wanted to make sure it was OK to do it ourselves here.

Reply:
Yep, that would be fine. At the API level, we want to leave validation to the user. However, at the libraries level, we are trying to save customers as much time as possible - fail fast with great error strings, so a person can fix the issue quickly.

Are we good to merge this PR in? (assuming the work you are talking about will be done in a different PR?)